src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -231,12 +233,12 @@ public AnalyzerOptions AnalyzerOptions
         {
             get
             {
-                if (_analyzerOptions == null)
+                if (_doNotAccessDirectlyAnalyzerOptions == null)
```

Comment:
we could, but even without it, it will return logically same thing if we ever hit race.